libflux/go/libflux/buffer.go

```diff
@@ -0,0 +1,26 @@
+// +build libflux
+
+package libflux
+
+// #cgo CFLAGS: -I.
+// #cgo LDFLAGS: -L. -llibstd
```

Comment:
I think you want to use `-lflux` here instead of `-llibstd`, since the function you're using (`flux_free`) is implemented in the `flux` crate.  I assume that this compiled, I wonder why it did..